### DISCUSSION TITLE
Having one PR per module per folders

### DISCUSSION THREAD

**Discussion by ArchiFleKs** [renovatebot/renovate#9513] [Request Help]
Hi,

I'm currently using renovate in the context of Terraform infrastructure folder. This works great with Helm and Terraform/Terragrunt module since the latest issue have been fixed.

One thing I'm trying to achieve is to have separate PR per folder per manager. If that is unclear, my folders are usually like this:

```
live/
  production
  staging
```

The issue I have right know if that the PR for one specific module are for both folder, if module A is update to v0.1 and this module is used in both production and staging folder I get a PR for all the files using this specific module. The issue with that is that I cannot merge in staging first, then in production. I'd like to get two per (for each used module):

* feat(tf-staging) : -> only the staging/* folder
* feat(tf-production): -> Only for the production/* folder

Is this feasible at all ? I guess another option is to play with group ?

**Comment by viceice** [comment]
Checkout https://docs.renovatebot.com/configuration-options/#additionalbranchprefix

**Reply by rarkins** [reply]
Related: https://github.com/renovatebot/renovate/discussions/7993

**Reply by ArchiFleKs** [reply]
Thanks it is working with this configuration:
```
      {
        "matchManagers": ["terraform", "terragrunt"],
        "semanticCommitScope": "tf-staging",
        "semanticCommitType": "feat",
        "matchPaths": ["staging"],
        "additionalBranchPrefix": "staging-"
      },
      {
        "matchManagers": ["terraform", "terragrunt"],
        "semanticCommitScope": "tf-production",
        "semanticCommitType": "feat",
        "matchPaths": ["production"],
        "additionalBranchPrefix": "production-"
      }
```

**Reply by ArchiFleKs** [reply]
Reopening, with this setup I have only one PR which keeps getting update, for example when update a module to v1.2, it will reopen the old PR for v1.1 and rename it. How could I have a new PR, by including the module version like it is done by default.

Here my branch are like this: 
* renovate/terraform-aws-modules-eks-aws-15.x : non match path
* renovate/production-particuleio-addons-kubernetes-1.x : reopen old PR

**Reply by ArchiFleKs** [reply]
In a more standard setup, without path matching, I'm getting two different PR with the same branch name:

* https://github.com/particuleio/terraform-kubernetes-addons/pull/109
* https://github.com/particuleio/terraform-kubernetes-addons/pull/118

**Reply by ArchiFleKs** [reply]
It might be just a race condition because the branch were not yet deleting when renovate run so it reopened the PR as weel as pushed in the old branch
